cocotb/share/lib/embed/gpi_embed.cpp

```diff
@@ -261,30 +256,22 @@ extern "C" COCOTB_EXPORT int _embed_sim_init(int argc,
     }
     // cocotb must hold _sim_event until _embed_sim_cleanup runs
 
-    // Build argv for cocotb module
-    auto argv_list = PyList_New(argc);
-    if (argv_list == NULL) {
-        // LCOV_EXCL_START
-        PyErr_Print();
-        return -1;
-        // LCOV_EXCL_STOP
-    }
+    std::vector<wchar_t *> argv_wchar;
+    DEFER({
+        for (auto arg : argv_wchar) {
+            PyMem_RawFree(arg);
+        }
+    })
     for (int i = 0; i < argc; i++) {
-        // Decode, embedding non-decodable bytes using PEP-383. This can only
-        // fail with MemoryError or similar.
-        auto argv_item = PyUnicode_DecodeLocale(argv[i], "surrogateescape");
-        if (!argv_item) {
-            // LCOV_EXCL_START
-            PyErr_Print();
+        auto arg = Py_DecodeLocale(argv[i], NULL);
```

Comment:
Does changing from `PyUnicode_DecodeLocale(argv[i], "surrogateescape")` to `Py_DecodeLocale(argv[i], NULL)` change cocotb behavior at all?
@eric-wieser 

From the docs for `PyUnicode_DecodeLocale()`:
> Changed in version 3.7: The function now also uses the current locale encoding for the `surrogateescape` error handler, except on Android. Previously, [Py_DecodeLocale()](https://docs.python.org/3/c-api/sys.html#c.Py_DecodeLocale) was used for the `surrogateescape`, and the current locale encoding was used for `strict`.

Reply:
I'm less convinced this commit is beneficial, looking over it now. The only real value in calling `PySys_SetArgv` is to set argv *before* initialization to assist in determining the Python executable and environment. I'll rebase it out tonight unless there are objections.